### ISSUE TITLE
Problem initializing viewer from search URLs

### ISSUE THREAD

**Issue by jabrah** [jhu-digital-manuscripts/rosa2#388]
Navigating to a page in a book, I then do a search across the AOR collection.

Older implementation would capture the search service, then render a history with hash (`#aor/search`) then have the relevant search parameters.

Now imagine using browser navigation to recreate the search. We have the URL: `?...#aor/search`. This URL then opens up the book list and initiates a search across the collection, even though this is not what actually happened. We do not have enough information to recreate the actual event. In order to do this, we would need to capture both the search service used for the search, and the source hash/path.

Possibility:
* Include the short name of the search service as a search parameter. 
  * This option would be ideal when dealing with links from an RMap representation, as the single link would be able to properly initialize the viewer. The `service` query param, perhaps, can be abbreviated to use the search services short name, as the URL hash does.
```   
?...&service=col#col/book/image
?...&service=col#col/book/opening
```
* Investigate the stored list of history states to determine the last state of the viewer before the search. This would require only changes to the history controller. This would also mean zero changes to our URL scheme. When fully recreating a state from the research finding, we would have to generate two history entries, one to set the book view, one to represent the search
  * With this option, a single link would not be able initialize the UI as one would expect. It would do fine for _generating_ research findings, however, opening search links from the research finding to get into the viewer will not work.
* Remove collection searches from the sidebar - Only allowing searching within the specific book you are looking at. Wouldn't need a new 'service' query parameter in this case, however we would likely still have to add the `/image` or `/opening` suffix to the URL hash
  * Basically a narrower version of the first option

**Comment by jabrah** [comment]
Short name of the search service is included as a URL query parameter. Search service is then basically guessed based on the short name, which seems to work with the JHU collections.
